test/mysql_tests.erl

```diff
@@ -208,23 +208,23 @@ unix_socket_test() ->
     
 connect_queries_failure_test() ->
     process_flag(trap_exit, true),
-    ?assertError(_Reason, mysql:start_link([{user, ?user}, {password, ?password},
-                                            {queries, ["foo"]}])),
+    {error, Reason} = mysql:start_link([{user, ?user}, {password, ?password},
+                                        {queries, ["foo"]}]),
     receive
-        {'EXIT', _Pid, normal} -> ok
+        {'EXIT', _Pid, Reason} -> ok
```

Comment:
This happens because in this case `mysql_conn:init/1` returns `{stop, Reason}`. OTP handles this by returning `{error, Reason}` in response to `gen_server:start_link` and exiting the `gen_server` process with the same `Reason`. In other words, it can't be different due to OTP internals.

So we can have either `initial_queries_failed` in both the error tuple and the exit message (downside, you don't get to know _what_ was wrong with your queries), or the plain reason like in my implementation, or a combination of both, but it will be always duplicated.

Reply:
In `change_user`, I'm emulating this behavior by returning the error tuple and stopping the connection process with the same reason, for consistency.

Reply:
OK! Thanks for the explanation!